keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -97,5 +107,26 @@ class DeliveryConfigController(
     validator.validate(deliveryConfig)
   }
 
-  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+  @Operation(
+    description = "Imports a delivery config manifest from source control."
+  )
+  @PostMapping(
+    path = ["/import"],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  fun import(
+    @RequestParam repoType: String,
+    @RequestParam projectKey: String,
+    @RequestParam repoSlug: String,
+    @RequestParam manifestPath: String,
+    @RequestParam ref: String?
+  ): DeliveryConfig {
+    val deliveryConfig =
+      importer.import(repoType, projectKey, repoSlug, manifestPath, ref ?: "refs/heads/master")
+
+    authorizationSupport.checkApplicationPermission(WRITE, APPLICATION, deliveryConfig.application)
+    authorizationSupport.checkServiceAccountAccess(SERVICE_ACCOUNT, deliveryConfig.serviceAccount)
+
+    return upsert(deliveryConfig)
+  }
```

Comment:
Something still needs to hit this endpoint then?

Reply:
Yes. I plan to refactor `orca` at some point to stop doing this work and just calling into this endpoint. For now, it's helped me troubleshoot and fix #1262.